storage/cache/subtree_cache.go

```diff
@@ -479,17 +426,17 @@ func (s *SubtreeCache) Flush(ctx context.Context, setSubtrees SetSubtreesFunc) e
 	return err
 }
 
-func (s *SubtreeCache) newEmptySubtree(id storage.NodeID, px []byte) *storagepb.SubtreeProto {
-	sInfo := s.stratumInfoForPrefixLength(id.PrefixLenBits)
+func (s *SubtreeCache) newEmptySubtree(id storage.NodeID) *storagepb.SubtreeProto {
```

Comment:
Some kind of assertion that the node id passed in is the root of a subtree would be worthwhile, if cheap. If this has an unbearable cost, then at least document that this is expected to be a subtree root.

Reply:
Done.

Reply:
Not sure that check is what was meant by @mhutchinson? At least in the map case it could possibly be stronger because of the non-uniform strata depths.  It may not be possible to apply it here though as it's not map specific code.

Reply:
Yeah, this is the most I could do here. The TODO that I added below is actually how it should be done - `SubtreeID` will be compliant with strata sizes because it will be generated by `TreeLayout`.